sycl/include/sycl/group_algorithm.hpp

```diff
@@ -843,7 +843,7 @@ joint_exclusive_scan(Group g, InPtr first, InPtr last, OutPtr result, T init,
     return ((v + divisor - 1) / divisor) * divisor;
   };
   typename std::remove_const<typename detail::remove_pointer<InPtr>::type>::type
-      x;
+      x = 0;
```

Comment:
I am not convinced we can rely on 0 always being a valid value for the resulting type. Could we maybe instead just zero-initialize it?
```suggestion
      x = {};
```

Reply:
Yes, you're probably right - your suggestion looks better to me!